app/utils/ListItem.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+object ListItem {
+
+  def createId[A](index: Int, item: A) = s"$index.${item.hashCode()}"
+
+  def findById[A](id: String, items: Seq[A]): Option[A] =
+    try {
+      id.split("\\.") match {
+        case Array(index: String, hashcode: String) if items(index.toInt).hashCode() == hashcode.toInt => Some(items(index.toInt))
+        case _                                                                                         => None
```

Comment:
Is this case necessary? Never sam the case without anything inside `case _`.

I think that without it you'll have an exception which will be handled in catch so you can remove it.

Reply:
Ah now I see how it looks like, formatting during review split this in few lines, so it wasn't readable